library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java

```diff
@@ -96,27 +143,28 @@ public PojoInnerJoin(int num, Class<?> outClass, String... keys)
     input.getClass().getDeclaredField(keys[index]);
 
     List<Map<String, Object>> curList = accu.get(index);
-    Map map = pojoToMap(input);
+    Map map = pojoToMap(input,index + 1);
     curList.add(map);
     accu.set(index, curList);
 
     return accu;
   }
 
-  private Map<String, Object> pojoToMap(Object input)
+  private Map<String, Object> pojoToMap(Object input, int streamIndex)
   {
     Map<String, Object> map = new HashMap<>();
+    List<KeyValPair<String,PojoUtils.Getter>>  gettersStream;
+    if (streamIndex == 1) {
+      gettersStream = gettersStream1;
+    } else {
+      gettersStream = gettersStream2;
+    }
```

Comment:
could be done in conditional way
List<KeyValPair<String,PojoUtils.Getter>> gettersStream = streamIndex == 1? gettersStream1 : getterStream2;